src/board.ts

```diff
@@ -334,6 +335,37 @@ export function getKeyAtDomPos(pos: cg.NumberPair, asWhite: boolean, bounds: Cli
   return (file >= 0 && file < 8 && rank >= 0 && rank < 8) ? pos2key([file, rank]) : undefined;
 }
 
+function isAlreadySnapped(orig: cg.Pos, targetKey: cg.Key | undefined): boolean {
```

Comment:
This mostly duplicates the `queen(...) || knight(...)` logic.

Reply:
`validSnapPos` previously did not have knight moves so I combined snapping knight moves into this short circuit. I'll remove it now